boto3/ec2/createtags.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+def inject_create_tags(event_name, class_attributes, **kwargs):
+    """This injects a custom create_tags method onto the ec2 service resource
+
+    This is needed because the resource model is not able to express
+    creating multiple tag resources based on the fact you can apply a set
+    of tags to multiple ec2 resources.
+    """
+    if event_name == 'creating-resource-class.ec2':
```

Comment:
Yeah it is needed. The creation of the service resource emits ``'creating-resource-class.ec2'`` while other resources like ``Instance`` emits ``'creating-resource-class.ec2.Instance'``. If I did not have the check, my custom ``create_tags()`` method would be added to every resource due to how the event system works.

The only other option would be to add a ``'creating-resource-class.ec2.ServiceResource'`` event when creating the service resource. Thoughts?